openmls/src/extensions/mod.rs

```diff
@@ -370,13 +440,49 @@ impl TryFrom<Vec<Extension>> for Extensions {
     }
 }
 
-impl Extensions {
-    fn find_by_type(&self, extension_type: ExtensionType) -> Option<&Extension> {
+impl<T: TypedExtension> Extensions<T> {
+    fn find_by_type(&self, extension_type: ExtensionType) -> Option<&T> {
         self.unique
             .iter()
             .find(|ext| ext.extension_type() == extension_type)
     }
+}
+
+impl Extensions<GroupContextExtension> {
```

Comment:
Hm, I don't like this. As long as we have only one that may be fine, but if we have all of the extensions in separate types, this will blow up. Any ideas for something better, without macros?